README.md

```diff
@@ -33,108 +34,242 @@ Menu
 
-(Note that if you use the Docker image, the command is slightly different: `docker run -i loadimpact/k6 run - <script.js`)
+For even more complex scenarios, you can use the k6 [REST API](https://docs.k6.io/docs/rest-api) and the `k6 status`, `k6 scale`, `k6 pause`, `k6 resume` CLI commands to manually control a running k6 test. For [cloud-based tests](https://docs.k6.io/docs/cloud-execution), executed on Load Impact's managed infrastructure via the `k6 cloud` command, you can also specify the VU distribution percentages for different load zones when executing load tests, giving you scalable and geographically-distributed test execution.
 
-For more information on how to get started running k6, please look at the [Running k6](https://docs.k6.io/docs/running-k6) documentation. If you want more info on the scripting API or results output, you'll find that also on [https://docs.k6.io](https://docs.k6.io).
 
----
+### Setup and teardown
+
+Beyond the init code and the required VU stage (i.e. the `default` function), which is code run for each VU, k6 also supports test wide setup and teardown stages, like many other testing frameworks and tools. The `setup` and `teardown` functions, like the `default` function, need to be `export`ed. But unlike the `default` function, `setup` and `teardown` are only called once for a test - `setup()` is called at the beginning of the test, after the init stage but before the VU stage (`default` function), and `teardown()` is called at the end of a test, after the last VU iteration (`default` function) has finished executing. This works even when executing a test in the distributed [cloud execution](https://docs.k6.io/docs/cloud-execution) mode via `k6 cloud`.
+
+See the [full example below](#example-code) for an illustration of how `setup()` and `teardown()` work, how they can be useful in a realistic situation, and how you can pass information from `setup()` to the `default` function and `teardown()`. You can also find more information and examples in the k6 docs [here](https://docs.k6.io/docs/test-life-cycle#section-setup-and-teardown-stages).
+
+
+### Metrics, tags and groups
+
+By default k6 measures and collects a lot of metrics about the things your scripts do - the duration of different script iterations, how much data was sent and received, how many HTTP requests were made, what was their duration, and even how long did the TLS handshake of a particual HTTPS request take. You see a summary of those built-in metrics in the output when you run the simplest possible k6 test, e.g. `k6 run github.com/loadimpact/k6/samples/http_get.js`. More information about the different built-in metrics collected by k6 (and how some of them can be accessed from inside of the scripts) can be seen in the docs [here](https://docs.k6.io/docs/result-metrics).
```

Comment:
By default k6 measures and collects a lot of metrics about the things your scripts do - the duration of different script iterations, how much data was sent and received, how many HTTP requests were made, the duration of those HTTP requests, and even how long did the TLS handshake of a particular HTTPS request take. To see a summary of these built-in metrics in the output, you can run a simple k6 test, e.g. `k6 run github.com/loadimpact/k6/samples/http_get.js`. More information about the different built-in metrics collected by k6 (and how some of them can be accessed from inside of the scripts) is available in the docs [here](https://docs.k6.io/docs/result-metrics).